src/Tools/_framework/Paths/Courses.jsx

```diff
@@ -94,15 +94,29 @@ export async function action({ request }) {
   }
 }
 
-export async function loader({ params }) {
-  const { data } = await axios.get("/api/getCoursePermissionsAndSettings.php");
-  return {
-    courses: data.permissionsAndSettings,
-  };
+export async function loader() {
+  try {
+    let success = true;
+    const { data } = await axios.get(
+      "/api/getCoursePermissionsAndSettings.php",
+    );
+
+    return {
+      success,
+      courses: data.permissionsAndSettings,
+    };
+  } catch (e) {
+    return { success: false, message: e.response.data.message };
+  }
 }
 
 export function Courses() {
-  const { courses } = useLoaderData();
+  const { success, message, courses } = useLoaderData();
+
+  if (!success) {
```

Comment:
I don't get the point of catching the exception in the loader, setting success to false and then detecting it is false and re-throwing an exception here. Why don't we just standardize the behavior to always throw an exception in the loader? I do understand until we get error codes everywhere this will mean we need to throw an exception when we get a 200 response code with "success=false", but it seems to make more sense to me to put that handling in some utility code that we use in all the loaders, rather than spread out this logic around a bunch of react components when they consume loaderData().

Reply:
We specifically want to send certain errors to the components.  It's by design.  There is likely a better design, but this is as far as we are right now.